gobblin-runtime/src/main/java/org/apache/gobblin/runtime/Task.java

```diff
@@ -1048,6 +1048,7 @@ boolean hasTaskFuture() {
   public synchronized boolean cancel() {
     LOG.info("Calling task cancel with interrupt flag: {}", this.shouldInterruptTaskOnCancel);
     if (this.taskFuture != null && this.taskFuture.cancel(this.shouldInterruptTaskOnCancel)) {
+      this.taskStateTracker.onTaskCommitCompletion(this);
```

Comment:
I'm not quite following why you add this now--please explain.  at this point, unclear whether a comment would help or whether it ought to be obvious (yet I'm not graping)

Reply:
We need to call this method to make the job as done, otherwise the thread will keep running and not know task has been finished https://github.com/apache/gobblin/blob/master/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java#L195

Reply:
a comment to that effect not be mandatory, but likely helpful

Reply:
Yeah, I just add comments in the new commit